### PR TITLE
Update content of flash messages for lists and list items

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description 

This adds additional content based on feedback from Design that the flash messages should be slightly more verbose and use consistent language.

The document with the requested changes can be found here https://docs.google.com/document/d/1FWW9oY6vD6WTJH5QJh2xqV2rc1onU9ajC63Bo2xUYno/edit

### New list 

|Before|After|
|-----------|-----------|
|image|image|

### Edit list

|Before|After|
|-----------|-----------|
|image|image|

### Delete list 

|Before|After|
|-----------|-----------|
|image|image|

### Adding links

|Before|After|
|-----------|-----------|
|image|image|

**and singular**

image
 
### Removing link

|Before|After|
|-----------|-----------|
|image|image|

### Reordering links

|Before|After|
|-----------|-----------|
|image|image|

### Moving link to different list

|Before|After|
|-----------|-----------|
|image|image|

## Trello card 

https://trello.com/c/9VoxrVPi/413-rebuild-curate-lists-page-on-collections-publisher

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Comment by nnagewad** [comment]
👍🏽

File: spec/controllers/lists_controller_spec.rb [hunk 1]
```diff
@@ -264,7 +276,7 @@
     context "Tag is a MainstreamBrowsePage" do
       let(:tag) { create(:mainstream_browse_page, :published) }
 
-      it "creates a new list item and makes the correct calls to the Publishing API" do
+      it "updates the list link ordering and makes the correct calls to the Publishing API" do
```
Comment (kevindew) [inline spec/controllers/lists_controller_spec.rb:279]: Nice fix on the terminology here.

File: app/controllers/lists_controller.rb [hunk 2]
```diff
@@ -122,7 +122,8 @@ def update_list_items
 
     if save_list_items
       ListPublisher.new(@tag).perform
-      flash[:notice] = "Items added to list successfully"
+      new_list_item_count = edit_list_items_params.count
+      flash[:notice] = "#{new_list_item_count} #{'link'.pluralize(new_list_item_count)} successfully added to the list"
```
Comment (kevindew) [inline app/controllers/lists_controller.rb:126]:
```suggestion
      flash[:notice] = "#{'pluralize(edit_list_items_params.count, 'link')} successfully added to the list"
```

minor, this seems a little cleaner and save the need for an extra variable.

File: spec/controllers/lists_controller_spec.rb [hunk 3]
```diff
@@ -110,16 +111,25 @@
...
-        patch :update_list_items, params: { tag_id: tag.content_id, id: list.id, list: { list_items: ["/new-list"] } }
+        patch :update_list_items, params: {
+          tag_id: tag.content_id,
+          id: list.id,
+          list: { list_items: ["/new-list", "/newer-list"] },
+        }
 
-        new_list_item = list.reload.list_items.last
+        new_list_item1 = list.reload.list_items[-2]
+        new_list_item2 = list.list_items.last
```
Comment (kevindew) [inline spec/controllers/lists_controller_spec.rb:121]: The `-2` here is a bit cryptic.

Assuming this is an active record collection I wonder if you can do: `new_list_item1, new_list_item2 = list.list_items.last(2)`

**Review by kevindew — COMMENTED** [review]
Nice changes, screenshots look great. I've just made a couple of small code suggestions.

File: spec/controllers/lists_controller_spec.rb (hunk 3 above)
Comment (davidgisbey) [reply spec/controllers/lists_controller_spec.rb:121]: Ooooh that's really cool. Thanks!

File: app/controllers/lists_controller.rb (hunk 2 above)
Comment (davidgisbey) [reply app/controllers/lists_controller.rb:126]: I did think about using this, but it requires inclusion of the `ActionView::Helpers::TextHelper` module. Do you think that's preferable?

Comment (kevindew) [reply app/controllers/lists_controller.rb:126]: Oh do we not have that already included in order to be able to call `pluralize` on a string? I think you can call helpers from a controller by doing `helpers.pluralize` with the helpers object having access to all your view helper methods (https://www.bigbinary.com/blog/rails-add-helpers-method-to-ease-usage-of-helper-modules-in-controllers)

**Comment by davidgisbey** [comment]
Have addressed the comments. Thanks very much for the review!

**Review by ollietreend — APPROVED** [review]
Nice – looks good to me 👍🏻

**Pull request merged**